primo/data_parser/well_data.py

```diff
@@ -866,10 +874,41 @@ def _process_input_data(self):
 
     def _append_fed_dac_data(self):
         """Appends federal DAC data"""
-        census_year = self.config.census_year
-        # TODO:
-        # Append Tract ID/GEOID, population density, Total population, land area,
-        # federal DAC score to the DataFrame
+        if len(self.data) == 0:
+            # Nothing to do if no data
+            return
+
+        state_code = identify_state(self)
+        census_tracts = get_state_census_tracts(state_code, DAC_TRACT_YEAR)
+        gdf = get_data_as_geodataframe(self)
+
+        # Spatial join to identify tract id associated with every well
+        # Per 2010 data
+        gdf = gdf.sjoin(census_tracts, how="left", predicate="within")
+        self.data["Census Tract ID [2010]"] = pd.to_numeric(gdf["GEOID10"])
+
+        cejst_data = get_cejst_data()
+        joined_data = self.data.merge(
+            cejst_data,
+            left_on="Census Tract ID [2010]",
+            right_on="Census tract 2010 ID",
+            how="left",
+        )
+        self.data["Percent area disadvantaged"] = joined_data[
+            "Percentage of tract that is disadvantaged by area"
+        ]
+
+        self.data["Percent pop disadvantaged"] = joined_data[
+            "Percentage of tract that is disadvantaged by area"
+        ]
+
+        self.data["Disadvantaged Score"] = (
+            self.data["Percent area disadvantaged"]
+            + self.data["Percent pop disadvantaged"]
+        ) / 2
```

Comment:
I'm assuming that columns `"Percent area disadvantaged"` and `"Percent pop disadvantaged"` are not needed. If that's the case, I suggest not adding them to the `data` DataFrame.
```python
fed_dac_data = (
    joined_data["Percentage of tract that is disadvantaged by area"] +
    joined_data["Percent pop disadvantaged"]
) / 2

self.add_new_column_ordered("fed_dac", "Federal DAC Data", fed_dac_data)
```
Also, I'm suggesting a change to the column name, because the score column will be named as "<column-name> Score [0-weight]". So, it will read as "Disadvantaged Score Score [0-weight]". With the suggested name, the score column's name will be "Federal DAC Data Score [0-weight]".

Reply:
Done